deno/src/main.ts

```diff
@@ -0,0 +1,208 @@
+    sha256: await sha256(metaText),
+    dest: `deno_dir/npm/registry.npmjs.org/${pkg.module}`,
+    "dest-filename": "registry.json",
+  };
+
+  const [checksumType, checksumValue] = splitOnce(
+    meta.versions[pkg.version].dist.integrity,
+    "-",
+  );
+  const pkgData: Record<string, unknown> = {
+    type: "archive",
+    "archive-type": "tar-gzip",
+    url:
+      `https://registry.npmjs.org/${pkg.module}/-/${pkg.name}-${pkg.version}.tgz`,
+    [checksumType]: base64ToHex(checksumValue),
+    dest: `deno_dir/npm/registry.npmjs.org/${pkg.module}/${pkg.version}`,
+  };
+
+  if (pkg.cpu) {
+    pkgData["only-arches"] = [pkg.cpu];
+  }
+
+  return [metaData, pkgData];
+}
+
+export async function main(lockPath: string, outputPath = "deno-sources.json") {
+  const lock = JSON.parse(Deno.readTextFileSync(lockPath));
+  if (lock.version !== "5") {
+    throw new Error(`Unsupported deno lock version: ${lock.version}`);
+  }
+
+  const jsrPkgs: Pkg[] = !lock.jsr ? [] : Object.keys(lock.jsr).map((pkg) => {
+    const r = splitOnce(pkg, "@", "right");
+    const name = r[0].split("/")[1];
+    return { module: r[0], version: r[1], name };
+  });
+  jsrPkgs;
+  const npmPkgs: Pkg[] = !lock.npm ? [] : Object.entries(lock.npm)
+    .filter((
+      // deno-lint-ignore no-explicit-any
+      [_key, val]: any,
+    ) => (val.os === undefined || val.os?.at(0) === "linux"))
+    // deno-lint-ignore no-explicit-any
+    .map(([key, val]: any) => {
+      const r = splitOnce(key, "@", "right");
+      const name = r[0].includes("/") ? r[0].split("/")[1] : r[0];
+      const cpu = val.cpu?.at(0);
+      return {
+        module: r[0],
+        version: r[1],
+        name,
+        cpu: cpu === "x64" ? "x86_64" : cpu === "arm64" ? "aarch64" : cpu,
+      };
+    });
+  //url: https://registry.npmjs.org/@napi-rs/cli/-/cli-2.18.4.tgz
+  npmPkgs;
+  const httpPkgsData = !lock.remote
+    ? []
+    : Object.entries(lock.remote).map(async ([urlStr, checksum]) => {
+      const url = new URL(urlStr);
+      const segments = await Promise.all(
+        urlSegments(url)
+          .map(async (part) => shouldHash(part) ? await shortHash(part) : part),
+      );
+      const filename = segments.pop();
+      return {
+        type: "file",
+        url: urlStr,
+        sha256: checksum,
+        dest: `vendor/${url.hostname}/${segments.join("/")}`,
+        "dest-filename": filename,
+      };
+    });
+
+  const flatpakData = [
+    await Promise.all(
+      jsrPkgs.map((pkg) => jsrPkgToFlatpakData(pkg)),
+    ).then((r) => r.flat()),
+    await Promise.all(npmPkgs.map((pkg) => npmPkgToFlatpakData(pkg))).then(
+      (r) => r.flat(),
+    ),
+    await Promise.all(httpPkgsData),
+  ].flat();
+  // console.log(flatpakData);
+  Deno.writeTextFileSync(
+    outputPath,
+    JSON.stringify(flatpakData, null, 2),
+  );
+}
+
+if (import.meta.main) {
+  const arg = Deno.args[0];
+  if (!arg) {
+    console.error("No argument provided");
+    Deno.exit(1);
```

Comment:
can you show a small usage hint text here?

Reply:
done